server/py/services/alerts/crud/alerts.py

```diff
@@ -94,8 +94,12 @@ def store_alert(
 
         self._validate_and_mask_notifications(alert_data)
 
-        new_alert = framework.utils.singletons.db.get_db().store_alert(
-            session, alert_data
+        new_alert = (
+            framework.utils.singletons.db.get_db().store_alert(session, alert_data)
+            if existing_alert
+            else framework.utils.singletons.db.get_db().create_alert(
+                session, alert_data
+            )
```

Comment:
Why? `store_alert` should handle that already.

Reply:
@TomerShor to avoid additional get_alert call when creating alert